src/crewai/agents/agent_builder/utilities/base_agent_tool.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+from pydantic import BaseModel, Field
+from crewai.agents.agent_builder.base_agent import BaseAgent
+from crewai.task import Task
+from crewai.utilities import I18N
+
+
+class BaseAgentTools(BaseModel, ABC):
+    """Default tools around agent delegation"""
+
+    agents: List[BaseAgent] = Field(description="List of agents in this crew.")
+    i18n: I18N = Field(default=I18N(), description="Internationalization settings.")
+
+    @abstractmethod
+    def tools(self):
+        pass
+
+    def _get_coworker(self, coworker: Optional[str], **kwargs) -> Optional[str]:
+        coworker = coworker or kwargs.get("co_worker") or kwargs.get("coworker")
+        if coworker and coworker.startswith("[") and coworker.endswith("]"):
+            coworker = coworker[1:-1].split(",")[0]
+        return coworker
+
+    def delegate_work(
+        self, task: str, context: str, coworker: Optional[str] = None, **kwargs
+    ):
+        """Useful to delegate a specific task to a coworker passing all necessary context and names."""
+        coworker = self._get_coworker(coworker, **kwargs)
+        return self._execute(coworker, task, context)
+
+    def ask_question(
+        self, question: str, context: str, coworker: Optional[str] = None, **kwargs
+    ):
+        """Useful to ask a question, opinion or take from a coworker passing all necessary context and names."""
+        coworker = self._get_coworker(coworker, **kwargs)
+        return self._execute(coworker, question, context)
+
+    def _execute(self, agent_role: Optional[str], task: str, context: str):
```

Comment:
<img width="705" alt="Screenshot 2024-06-26 at 11 59 52 PM" src="https://github.com/joaomdmoura/crewAI/assets/109994880/7b9051c4-d907-4b1b-a0c9-3d4b1e4e7899">

We need to add back in the clean up that's in this image to support weaker models.

Reply:
main does not have these changes. think we can this later ?